prow/plank/controller.go

```diff
@@ -77,7 +100,88 @@ func (c *Controller) Sync() error {
 	}
 }
 
-func (c *Controller) syncJob(pj kube.ProwJob, pm map[string]kube.Pod) error {
+func (c *Controller) syncJenkinsJob(pj kube.ProwJob) error {
+	var jerr error
+	if pj.Complete() {
+		return nil
+	} else if pj.Status.State == kube.TriggeredState {
+		// Start the Jenkins job.
+		pj.Status.State = kube.PendingState
+		br := jenkins.BuildRequest{
+			JobName: pj.Spec.Job,
+			Refs:    pj.Spec.Refs.String(),
+			BaseRef: pj.Spec.Refs.BaseRef,
+			BaseSHA: pj.Spec.Refs.BaseSHA,
+		}
+		if len(pj.Spec.Refs.Pulls) == 1 {
+			br.Number = pj.Spec.Refs.Pulls[0].Number
+			br.PullSHA = pj.Spec.Refs.Pulls[0].SHA
+		}
+		if build, err := c.jc.Build(br); err != nil {
+			jerr = fmt.Errorf("error starting Jenkins job: %v", err)
+			pj.Status.CompletionTime = time.Now()
+			pj.Status.State = kube.ErrorState
+			pj.Status.URL = testInfra
+		} else {
+			pj.Status.JenkinsQueueURL = build.QueueURL.String()
+			pj.Status.JenkinsBuildID = build.ID
+			pj.Status.JenkinsEnqueued = true
+		}
+		if err := c.report(pj); err != nil {
+			return fmt.Errorf("error reporting to crier: %v", err)
+		}
+	} else if pj.Status.JenkinsEnqueued {
+		if eq, err := c.jc.Enqueued(pj.Status.JenkinsQueueURL); err != nil {
+			jerr = fmt.Errorf("error checking queue status: %v", err)
+			pj.Status.JenkinsEnqueued = false
+			pj.Status.CompletionTime = time.Now()
+			pj.Status.State = kube.ErrorState
+			pj.Status.URL = testInfra
+		} else if eq {
+			// Still in queue.
+			return nil
+		} else {
+			pj.Status.JenkinsEnqueued = false
+		}
+	} else if status, err := c.jc.Status(pj.Spec.Job, pj.Status.JenkinsBuildID); err != nil {
+		jerr = fmt.Errorf("error checking build status: %v", err)
+		pj.Status.CompletionTime = time.Now()
+		pj.Status.State = kube.ErrorState
+		pj.Status.URL = testInfra
```

Comment:
when do we need to report to crier?  the ReportToCrier func is not documented :-(